### ISSUE TITLE
binance.roundStep issue

### ISSUE THREAD

**Issue by naksi** [jaggedsoft/node-binance-api#132]
```
let stepSize = 0.00100000;
let amount = 0.1023;

amount = binance.roundStep(amount, stepSize);

console.log(amount)  //0.10200000000000001
```

**Comment by jaggedsoft** [comment]
Hi there, thank you for the report!
This has been fixed in v0.4.20 https://github.com/jaggedsoft/node-binance-api/commit/01787780472baf32463afbdb955e1e847521f924
roundStep will now set the precision for you, so extra action is necessary


You would have to count the decimal places to get the precision:
```js
let stepSize = 0.00100000;
let precision = stepSize.toString().split(".")[1].length || 0; // easier version: binance.getPrecision(number)
let amount = 0.1023;
amount = binance.roundStep(amount, stepSize);
console.log("amount: ", amount.toFixed(precision));  //0.102
```

I'd like to build all of this into the library so it's done for you. my solution for now:
1. Run exchangeInfo at program startup
2. Save this data to some global scope, like `global.filters` or `let assetInfo = {};`
3. Apply roundStep with stepSize `binance.roundStep(amount, stepSize);`

```js
//minQty = minimum order quantity
//minNotional = minimum order value (price * quantity)
binance.exchangeInfo(function(error, data) {
	let minimums = {};
	for ( let obj of data.symbols ) {
		let filters = {status: obj.status};
		for ( let filter of obj.filters ) {
			if ( filter.filterType == "MIN_NOTIONAL" ) {
				filters.minNotional = filter.minNotional;
			} else if ( filter.filterType == "PRICE_FILTER" ) {
				filters.minPrice = filter.minPrice;
				filters.maxPrice = filter.maxPrice;
				filters.tickSize = filter.tickSize;
			} else if ( filter.filterType == "LOT_SIZE" ) {
				filters.stepSize = filter.stepSize;
				filters.minQty = filter.minQty;
				filters.maxQty = filter.maxQty;
			}
		}
		//filters.baseAssetPrecision = obj.baseAssetPrecision;
		//filters.quoteAssetPrecision = obj.quoteAssetPrecision;
		filters.orderTypes = obj.orderTypes;
		filters.icebergAllowed = obj.icebergAllowed;
		minimums[obj.symbol] = filters;
	}
	console.log(minimums);
	global.filters = minimums;
	//fs.writeFile("minimums.json", JSON.stringify(minimums, null, 4), function(err){});
});
```
![image](https://user-images.githubusercontent.com/4283360/36242753-725126de-11d2-11e8-9d03-5ce0d20a62cd.png)


Then you could do this:
```js
const symbol = 'BNBBTC';
let amount = 0.1023;
let stepSize = global.filters[symbol].stepSize;
amount = binance.roundStep(amount, stepSize);
```

**Comment by naksi** [comment]
Thank you for the amazing library, I appreciate all the work you and other contributors have put into this!
I was previously storing exchangeInfo filter data in a file and then reading from it when needed, but It's definitely better idea and also much faster to read from global variable.
